src/lib.rs

```diff
@@ -273,6 +273,11 @@ enum IpfsEvent {
     PubsubPublish(String, Vec<u8>, OneshotSender<()>),
     PubsubPeers(Option<String>, OneshotSender<Vec<PeerId>>),
     PubsubSubscribed(OneshotSender<Vec<String>>),
+    GossipsubSubscribe(String, OneshotSender<Option<SubscriptionStream>>),
+    GossipsubUnsubscribe(String, OneshotSender<bool>),
+    GossipsubPublish(String, Vec<u8>, OneshotSender<()>),
+    GossipsubPeers(Option<String>, OneshotSender<Vec<PeerId>>),
+    GossipsubSubscribed(OneshotSender<Vec<String>>),
```

Comment:
Last I looked, gossipsub is a superset of the floodsub so I think the old entries in IpfsEvent and generally any api just should switch to using gossipsub as the backend. At least it should spare you from writing the boilerplate methods.